### ISSUE TITLE
Generated specific records do not handle ByteBuffer correctly in put

### ISSUE THREAD

**Issue by tmoschou** [julianpeeters/avrohugger#127]
The generated specific records have errors regarding setting a field of type `bytes`.

The generated setter is

```scala
  def put(field$: Int, value: Any): Unit = {
    (field$: @switch) match {
      case 0 => this.fieldName = {
        value match {
          case (buffer: java.nio.ByteBuffer) => {
            buffer.array()
          }
        }
      }.asInstanceOf[Array[Byte]]
      // ...
      case _ => new org.apache.avro.AvroRuntimeException("Bad index")
    }
    ()
  }
```

* Firstly `ByteBuffer.array()` is an optional operation, that can throw `UnsupportedOperationException` If this buffer is not backed by an accessible array, e.g. by `ByteBuffer.allocateDirect()`
* Secondly, It can throw `ReadOnlyBufferException` If this buffer is backed by an array but is read-only
* Thirdly if the array is accessible, the code assumes that offset/position are `0` and limit/capacity are `array.length` for which assumptions should not be made. 

The bytes should be copied as such

```scala
val dup = buffer.duplicate()
val array = new Array[Byte](dup.remaining)
dup.get(array)
```

**Comment by julianpeeters** [comment]
Thank you @tmoschou! I expect to have some time for a fix this weekend or next

**Comment by julianpeeters** [comment]
Thanks again @tmoschou , this is fixed in avrohugger 1.0.0-RC22 and sbt-avrohugger 2.0.0-RC22
